src/reducers/sources.js

```diff
@@ -286,12 +291,39 @@ export function getSourceFromId(state: OuterState, id: string): Source {
   return getSourcesState(state).sources[id];
 }
 
-export function getSourceByURL(state: OuterState, url: string): ?Source {
-  return getSourceByUrlInSources(getSources(state), getUrls(state), url);
+export function getSourceByURL(
+  state: OuterState,
+  url: string,
+  isOriginal: boolean = false
+): ?Source {
+  // Pretty sources should always be original
+  if (isPrettyURL(url)) {
+    isOriginal = true;
+  }
+
+  return getSourceByUrlInSources(
+    getSources(state),
+    getUrls(state),
+    url,
+    isOriginal
```

Comment:
I think it would be nice to pass in the correct isOriginal flag so that it includes formatted


```diff
diff --git a/src/actions/tabs.js b/src/actions/tabs.js
index 53060be..9b138e8 100644
--- a/src/actions/tabs.js
+++ b/src/actions/tabs.js
@@ -9,7 +9,7 @@
  * @module actions/tabs
  */

-import { isOriginalId } from "devtools-source-map";
+import { isOriginal } from "../utils/source";

 import { removeDocument } from "../utils/editor";
 import { selectSource } from "./sources";
@@ -27,7 +27,7 @@ import type { Source } from "../types";

 export function updateTab(source: Source, framework: string): Action {
   const { url } = source;
-  const isOriginal = isOriginalId(source.id);
+  const isOriginal = isOriginal(source.id);

   return {
     type: "UPDATE_TAB",
@@ -39,7 +39,7 @@ export function updateTab(source: Source, framework: string): Action {

 export function addTab(source: Source): Action {
   const { url } = source;
-  const isOriginal = isOriginalId(source.id);
+  const isOriginal = isOriginal(source.id);

   return {
     type: "ADD_TAB",
diff --git a/src/utils/source.js b/src/utils/source.js
index 1837d4c..defbca4 100644
--- a/src/utils/source.js
+++ b/src/utils/source.js
@@ -87,6 +87,10 @@ export function isJavaScript(source: Source): boolean {
   );
 }

+export function isOriginal(source: Source) {
+  return isOriginalId(source.id) || isPrettyURL(source.url)
+}
+
```